onnxruntime/core/optimizer/nhwc_transformer.cc

```diff
@@ -14,6 +15,208 @@ using namespace onnx_layout_transformation;
 
 namespace onnxruntime {
 
+/**
+ * @brief For identifying layout sensive operators
+ * as candidates for transforming to NHWC ops.
+*/
+struct OpIdInfo{
+  const std::string optype_;
+  const std::string domain_;
+  const api::DataType data_type_;
+
+  OpIdInfo(
+    const std::basic_string_view<char>& op,
+    const std::basic_string_view<char>& domain,
+    api::DataType data_type
+  ) : optype_(op), domain_(domain), data_type_(data_type)
+  {}
+
+  bool operator==(const OpIdInfo& other) const{
+    return optype_ == other.optype_ && domain_ == other.domain_ && data_type_ == other.data_type_;
+  }
+};
+
+/**
+ * @brief Hash function for \ref OpIdInfo
+*/
+class OpIdHash{
+  public:
+  size_t operator()(const OpIdInfo& op) const
+  {
+    size_t h1 = std::hash<std::string>{}(op.optype_);
+    size_t h2 = std::hash<std::string>{}(op.domain_);
+    size_t h3 = size_t(op.data_type_);
+    return h2 ^ (h1 << 4) ^ (h3 << 16);
+  }
+};
+
+/**
+ * @brief Information needed for operator layout transformation
+*/
+struct OpTransformInfo{
+  const std::string optype_;
+  const std::string domain_;
+  const int version_;
+  const bool has_channels_last_attrib_;
+};
+
+/**
+ * Constructing a mapping table to identify operators that need
+ * to be transformed, and map them to the new operators with
+ * NHWC format
+ */
+std::unordered_map<OpIdInfo, OpTransformInfo, OpIdHash> conv_table;
```

Comment:
>conv_table [](http://example.com/codeflow?start=56&length=10)

It will not be release after transform. Make it a member variable of NhwcTransformer?

Reply:
then I have to move whole bunch of stuff to header file, ahh.